frontend/src/components/shared/Button/styled-components.ts

```diff
@@ -0,0 +1,204 @@
+import { MouseEvent, ReactNode } from "react"
+import styled, { CSSObject } from "@emotion/styled"
+import { Theme } from "theme"
+import { transparentize } from "color2k"
+
+export enum Kind {
+  PRIMARY = "primary",
+  SECONDARY = "secondary",
+  LINK = "link",
+  ICON = "icon",
+  MINIMAL = "minimal",
+}
+
+export enum Size {
+  XSMALL = "xsmall",
+  SMALL = "small",
+  MEDIUM = "medium",
+  LARGE = "large",
+}
+
+export interface ButtonProps {
+  kind: Kind
+  size?: Size
+  onClick?: (event: MouseEvent<HTMLButtonElement>) => any
```

Comment:
I see this is coming from what we currently have but I find it very odd that we would have a button with an optional onClick...